MISRA.md

```diff
@@ -1,40 +1,136 @@
-| Rule 14.4 | Required | Condition `0` not having an essentially boolean type is used in portable layer of the Kernel for `taskENTER_CRITICAL` and `taskEXIT_CRITICAL`. |
-| Rule 8.6 | Required | Some functions are declared but not defined since they are to be defined by the user application to provide 'hooks' for certain events. |
-| Rule 2.3 | Advisory |  |
-| Rule 20.10 | Advisory | For casting between pointer types, FreeRTOS-Plus-TCP uses utility inline functions created using the `##` operator to improve readability. |
-| Rule 10.4 | Required | Signed and unsigned variables are used together due to the variables acquired from the portable code. |
-| Rule 13.5 | Required | In `prvTCPSendPacket` (present in FreeRTOS_TCP_IP.c) persistent side effect of function call as right-hand operators of `||` has been verified to behave as expected. In FreeRTOS_Sockets.c (in function `FreeRTOS_sendto`) the violation noted is false positive. |
-| Rule 15.5 | Advisory | Some FreeRTOS-Plus-TCP functions use multiple `return` statements; These have not been refactored to comply with this rule. |
-| Rule 18.3 | Required | Relational operators are used with members of unions. |
-| Rule 21.6 | Required | In `FreeRTOS_strerror_r` (present in FreeRTOS_IP.c) `snprintf` is used to format strings before placing them into buffers. |
-| Rule 10.8 | Required | Unsigned expression in function `prvSendData` (present in FreeRTOS_TCP_IP.c) is cast to signed to compare with a signed variable. |
-| Rule 11.1 | Required | In `FreeRTOS_setsockopt` (present in FreeRTOS_Sockets.c), `void *` is used to pass variables of various types to be set as the socket option. One of these options allow the users to pass function pointer as well thereby violating the rule. To adhere to the Berkeley standard, the function signature cannot be changed. |
-| Rule 14.3 | Required | Use of conditional compilation hides the update to the controlling expression which can causes false positives of this rule. The violation occurs in the function prvProcessIPPacket present in FreeRTOS_IP.c. |
-| Rule 2.2 | Required | FreeRTOS+TCP uses unions which can cause false positives of this rule. The false positive occurs once in FreeRTOS_IP.c module in the function usGenerateChecksum. |
-| Rule 5.9 | Advisory | Utility functions are defined in various files to not break various other projects. |
-| Rule 8.9 | Advisory | For ease, configuration parameters are defined at a global scope even when used only once. |
-
+guidelines, with the deviations listed below. Compliance is checked with Coverity static analysis.
+Since the FreeRTOS-Plus-TCP library is designed for small-embedded devices, it needs to have a very small memory footprint and has to
+be efficient. To achieve that and to increase the performace of the IP-stack, it deviates from some MISRA rules.
+The project deviations, refer to [MISRA configuration file](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/tools/coverity_misra.config).
```

Comment:
"The project deviations, refer to.." doesn't quite make grammatical sense.  Maybe you intended "For project wide deviations, ref to..."?  In any case "Additionally, [link] contains project wide deviations" may be simpler.

Reply:
Agreed, updated.